core/src/smartcontracts/isi/world.rs

```diff
@@ -374,26 +371,26 @@ pub mod isi {
             let world = wsv.world_mut();
             let _ = world.upgraded_validator.insert(new_validator);
 
-            // let old_permission_token_definitions = wsv
-            //     .permission_token_definitions()
-            //     .values()
-            //     .cloned()
-            //     .collect::<HashSet<_>>();
-            // let new_permission_token_definitions =
-            //     HashSet::from_iter(new_permission_token_definitions);
-
-            // old_permission_token_definitions
-            //     .difference(&new_permission_token_definitions)
-            //     .map(|definition| Unregister::<PermissionTokenDefinition> {
-            //         object_id: definition.id.clone(),
-            //     })
-            //     .try_for_each(|unregister| unregister.execute(authority, wsv))?;
-
-            // new_permission_token_definitions
-            //     .difference(&old_permission_token_definitions)
-            //     .cloned()
-            //     .map(|definition| Register::<PermissionTokenDefinition> { object: definition })
-            //     .try_for_each(|new| new.execute(authority, wsv))?;
+            let old_permission_token_definitions = wsv
+                .permission_token_definitions()
+                .values()
+                .cloned()
+                .collect::<HashSet<_>>();
+            let new_permission_token_definitions =
+                HashSet::from_iter(new_permission_token_definitions);
```

Comment:
I think it's possible that `new_permission_token_definitions` will contain duplicates which will silently be discarded. It would be better if you used `HashSet::insert` and return error if a duplicate is found. There is no guarantee that there will not be two permission token structs of the same name